internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -272,8 +272,13 @@ func validate(csv *operatorsv1alpha1.ClusterServiceVersion) error {
 			log.Errorf("ClusterServiceVersion validation: [%s] %s", e.Type, e.Detail)
 		}
 		if r.HasError() {
-			return errors.New("got ClusterServiceVersion validation errors")
+			hasErrors = true
 		}
 	}
```

Comment:
```suggestion
	results := validation.ClusterServiceVersionValidator.Validate(csv)
	for _, r := range results {
		for _, w := range r.Warnings {
			log.Warnf("ClusterServiceVersion validation: [%s] %s", w.Type, w.Detail)
		}
		for _, e := range r.Errors {
		        hasErrors = true
			log.Errorf("ClusterServiceVersion validation: [%s] %s", e.Type, e.Detail)
		}
	}
```

Reply:
has no need to `if r.HasError() {`

Reply:
This is doing effectively the same thing as checking `r.HasError()`, but with a slight performance loss because it's resetting memory for each error (assuming the compiler doesn't optimize this into one operation).

Reply:
I  think here`slight performance loss` does not justify a would be more valuable looking for readability. However, 👍